File: Library.Api/ApiEndpoints.cs
namespace Library.Api;

public class ApiEndpoints
{
    private const string ApiBase = "api";

    public static class Author
    {
        private const string Base = $"{ApiBase}/authors";

        public const string Create = Base;

        public const string Get = $"{Base}/{{id:guid}}";

        public const string GetAll = Base;

        public const string Update = $"{Base}/{{id:guid}}";

        public const string Delete = $"{Base}/{{id:guid}}";

        public const string GetBooks = $"{Base}/{{id:guid}}/books";
    }

    public static class Book
    {
        private const string Base = $"{ApiBase}/books";

        public const string Create = Base;

        public const string Get = $"{Base}/{{idOrIsbn}}";

        public const string GetAll = Base;

        public const string Update = $"{Base}/{{id:guid}}";

        public const string Delete = $"{Base}/{{id:guid}}";

        public const string GetImage = $"{Base}/{{idOrIsbn}}/image";
    }

    public static class Auth
    {
        private const string Base = $"{ApiBase}/auth";

        public const string Register = $"{Base}/register";

        public const string Login = $"{Base}/login";

        public const string Refresh = $"{Base}/refresh";
    }

    public static class UserBook
    {
        private const string Base = $"{ApiBase}/me/books";

        public const string Create = Base;

        public const string GetAll = Base;

        public const string Delete = $"{Base}/{{bookId:guid}}";
    }
}

File: library.client/src/api/ApiEndpoints.js
const API_BASE_URL = process.env.REACT_APP_API_BASE_URL; // Base api URL is here

export const ApiEndpoints = {
  Author: {
    Create: `${API_BASE_URL}/authors`,
    Get: (id) => `${API_BASE_URL}/authors/${id}`,
    GetAll: (page, pagesize) => `${API_BASE_URL}/authors?pagesize=${pagesize}&page=${page}`,
    Update: (id) => `${API_BASE_URL}/authors/${id}`,
    Delete: (id) => `${API_BASE_URL}/authors/${id}`,
    GetBooks: (id) => `${API_BASE_URL}/authors/${id}/books`,
  },
  Book: {
    Create: `${API_BASE_URL}/books`,
    Get: (idOrIsbn) => `${API_BASE_URL}/books/${idOrIsbn}`,
    GetAll: (page, pagesize, title, genre, author) => `${API_BASE_URL}/books?pagesize=${pagesize}&page=${page}&title=${title}&genre=${genre}&author=${author}`,
    Update: (id) => `${API_BASE_URL}/books/${id}`,
    Delete: (id) => `${API_BASE_URL}/books/${id}`,
    GetImage: (idOrIsbn) => `${API_BASE_URL}/books/${idOrIsbn}/image`,
  },
  Auth: {
    Register: `${API_BASE_URL}/auth/register`,
    Login: `${API_BASE_URL}/auth/login`,
    Refresh: `${API_BASE_URL}/auth/refresh`,
  },
  UserBook: {
    Create: `${API_BASE_URL}/me/books`,
    GetAll: `${API_BASE_URL}/me/books`,
    Delete: (bookId) => `${API_BASE_URL}/me/books/${bookId}`,
  },
};

export default ApiEndpoints;